common/authenticator-requests/src/v5/registration.rs

```diff
@@ -143,24 +143,22 @@ impl GatewayClient {
         private_ips: IpPair,
         nonce: u64,
     ) -> Self {
-        // convert from 1.0 x25519-dalek private key into 2.0 x25519-dalek
-        #[allow(clippy::expect_used)]
-        let static_secret = x25519_dalek::StaticSecret::from(local_secret.to_bytes());
-        let local_public: x25519_dalek::PublicKey = (&static_secret).into();
+        let local_public = PublicKey::from(local_secret);
```

Comment:
while you're there, mind fixing that conversion everywhere? we're using `dalek` 2.0 everywhere now so it's quite reduntant